From 8d065fb7485466050b6876bb5ab1bab0794f7812 Mon Sep 17 00:00:00 2001
From: Samuel Verschelde <stormi-xcp@ylix.fr>
Date: Fri, 5 May 2023 20:09:58 +0200
Subject: [PATCH 5/5] Update xcp-ng.repo for the new repository structure

* Add the new incoming, ci, candidates and lab repositories.
* Add explanations for each repository.

Signed-off-by: Samuel Verschelde <stormi-xcp@ylix.fr>
---
 xcp-ng.repo | 47 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 47 insertions(+)

diff --git a/xcp-ng.repo b/xcp-ng.repo
index 0259a92..670bba9 100644
--- a/xcp-ng.repo
+++ b/xcp-ng.repo
@@ -1,3 +1,4 @@
+# Base packages. Frozen at the release of XCP-ng 8.2.0.
 [xcp-ng-base]
 name=XCP-ng Base Repository
 baseurl=http://mirrors.xcp-ng.org/8/8.2/base/x86_64/ http://updates.xcp-ng.org/8/8.2/base/x86_64/
@@ -6,6 +7,7 @@ gpgcheck=1
 repo_gpgcheck=1
 gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
 
+# Stable updates.
 [xcp-ng-updates]
 name=XCP-ng Updates Repository
 baseurl=http://mirrors.xcp-ng.org/8/8.2/updates/x86_64/ http://updates.xcp-ng.org/8/8.2/updates/x86_64/
@@ -14,6 +16,20 @@ gpgcheck=1
 repo_gpgcheck=1
 gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
 
+
+# The four repositories below this line are involved in the update build and testing process.
+# The typical flow is: incoming -> ci -> testing -> candidates -> updates
+
+# Updates set for imminent xcp-ng-updates publication upon completion of final testing.
+[xcp-ng-candidates]
+name=XCP-ng Candidates Repository
+baseurl=http://mirrors.xcp-ng.org/8/8.2/candidates/x86_64/ http://updates.xcp-ng.org/8/8.2/candidates/x86_64/
+enabled=0
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
+
+# Updates which are open for user feedback. They already passed internal CI.
 [xcp-ng-testing]
 name=XCP-ng Testing Repository
 baseurl=http://mirrors.xcp-ng.org/8/8.2/testing/x86_64/ http://updates.xcp-ng.org/8/8.2/testing/x86_64/
@@ -22,3 +38,34 @@ gpgcheck=1
 repo_gpgcheck=1
 gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
 
+# Updates undergoing CI tests. Will then move to xcp-ng-testing once ready.
+[xcp-ng-ci]
+name=XCP-ng CI Repository
+baseurl=http://mirrors.xcp-ng.org/8/8.2/ci/x86_64/ http://updates.xcp-ng.org/8/8.2/ci/x86_64/
+enabled=0
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
+
+# Updates are built here and moved to xcp-ng-ci for testing once ready.
+[xcp-ng-incoming]
+name=XCP-ng Incoming Repository
+baseurl=http://mirrors.xcp-ng.org/8/8.2/incoming/x86_64/ http://updates.xcp-ng.org/8/8.2/incoming/x86_64/
+enabled=0
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
+
+
+# "Lab" updates are not intended to become stable updates in the future
+# and are provided without any guarantees.
+# They may include version updates to system components that add interesting
+# features, but given the inherent risk associated with any version update,
+# they may not be suitable for everyone.
+[xcp-ng-lab]
+name=XCP-ng Lab Repository
+baseurl=http://mirrors.xcp-ng.org/8/8.2/lab/x86_64/ http://updates.xcp-ng.org/8/8.2/lab/x86_64/
+enabled=0
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
-- 
2.30.2

